### ISSUE TITLE
[Thinker 3.1] Crashes when saving

### ISSUE THREAD

**Issue by kernhe** [induktio/thinker#41]
Remark: run game in wine under linux

System tells me, that I have Iron Man option -- which is not true -- and then crashes.
Same behavior when I have Iron Man selected.

**Comment by induktio** [comment]
Ok, I have not seen this issue reported before. Can you provide crash logs and/or stack traces from wine? It should have detailed reporting features for that. Need to have more details on this issue.

**Comment by kernhe** [comment]
Thinker 3.1 is installed, as I can see it in the Alt-T dialog.

I run smac in a separate window:  wine explorer /desktop=SMAC,1400x960 thinker.exe ... and attached you'll find the stracktrace from wine.

[backtrace.txt](https://github.com/induktio/thinker/files/8507082/backtrace.txt)

BTW: the save file is corrupt, and takes 158.2 MiBs, not the usual 500KiBs.

**Comment by induktio** [comment]
So apparently the game crashes on page fault while writing to an invalid memory location. The instruction causing this is at 0x5a7e82 in game_data function which deals with reading/writing save games. This might be some kind of an original game bug because neither Thinker nor Scient's patch should affect that location.

From the logs it looks like you have Pracx also enabled but it probably isn't causing this. Can you confirm if the crash happens when you use a) official terranx.exe version, b) terranx.exe patched with Pracx only?

**Comment by kernhe** [comment]
TL;DR
Sorry for the trouble, you can close the ticket!

Longer feedback:

I did a thorough tryout, I even checked different combinations:
- with/without prax
- different versions of thinker mod (back to 1.0)
- from smax gog or from my CDs (yes, I have an old box and a cd drive)
- in linux or in virtualbox with windows

The issue was wine version 5.0, because in the virtualbox with an old Windows XP every combination worked fine, and when I tried the same installation (same files, because they were shared from linux to virtualbox) under wine it crashed as mentioned above.

I switched to winehq version 7.0.1 ... and it works as fine as it should run!
